base/setup/lib/fsutil.c

```diff
@@ -661,6 +690,66 @@ InstallBtrfsBootCode(
     return Status;
 }
 
+NTSTATUS
+InstallNtfsBootCode(
+    IN PCWSTR SrcPath,          // NTFS bootsector source file (on the installation medium)
+    IN HANDLE DstPath,          // Where to save the bootsector built from the source + partition information
+    IN HANDLE RootPartition)    // Partition holding the (old) NTFS information
+{
+    NTSTATUS Status;
+    UNICODE_STRING Name;
+    IO_STATUS_BLOCK IoStatusBlock;
+    LARGE_INTEGER FileOffset;
+    BOOTCODE OrigBootSector = {0};
+    BOOTCODE NewBootSector  = {0};
+
+    /* Allocate and read the current original partition bootsector */
+    Status = ReadBootCodeByHandle(&OrigBootSector, RootPartition, NTFS_BOOTSECTOR_SIZE);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("InstallNtfsBootCode: Status %lx\n", Status);
+        return Status;
+    }
+
+    /* Allocate and read the new bootsector (16 sectors) from SrcPath */
+    RtlInitUnicodeString(&Name, SrcPath);
+    Status = ReadBootCodeFromFile(&NewBootSector, &Name, NTFS_BOOTSECTOR_SIZE);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("InstallNtfsBootCode: Status %lx\n", Status);
+        FreeBootCode(&OrigBootSector);
+        return Status;
+    }
+
+    /* Adjust the bootsector (copy a part of the NTFS BPB) */
+    RtlCopyMemory(&((PNTFS_BOOTSECTOR)NewBootSector.BootCode)->OEMID,
+                  &((PNTFS_BOOTSECTOR)OrigBootSector.BootCode)->OEMID,
+                  FIELD_OFFSET(NTFS_BOOTSECTOR, BootStrap) - FIELD_OFFSET(NTFS_BOOTSECTOR, OEMID));
```

Comment:
```suggestion
                  sizeof(&((PNTFS_BOOTSECTOR)NewBootSector.BootCode)->OEMID));
```

Reply:
here the PR author wants to copy everything that is present from the beginning of the OEMID field + all the data that has been already set by the formatting routine, that follows the OEMID field (e.g. bytes per sector, ... MFT location fields, ...) up to but excluding the contents of the BootStrap field and following, into the new bootsector data, before copying it back into the partition.